### ISSUE TITLE
How to run the simulation

### ISSUE THREAD

**Issue by Aleksej9521** [AsdoubleU/Go2_Gazebo_Environment#1]
Hi, interesting repository :) I'm starting to work with the go2 robot and I want to test your work. How can I run the simulation? Just for curiosity, where did you find information about torque control for unitree robots?

**Comment by AsdoubleU** [comment]
Sorry for my answer is late. You can run this repository with following steps.

1. Build the package with "catkin build" command.
2. Add the packge file with ""source ~/go2_ws/devel/setup.bash" command.
3. Launch the gazebo environment of go2 robot with "roslaunch go2_gazebo go2_ground.launch" command for robot on the ground or  "roslaunch go2_gazebo go2_air.launch" command for robot hanging on the air.
4. You can run the algorithmns with the "roslaunch go2_torque torque.launch" command.

I haven't add the algorithmns for the robot walking or running, so you can only see the robot operate some pose.

And you can access the example codes at the unitree github.
https://github.com/unitreerobotics/unitree_sdk2

Thank you for read my codes!

**Comment by Aleksej9521** [comment]
Thank you for all the informations, I will try :)

**Comment by Aleksej9521** [comment]
Hi, I'm re-opening this issue just for a little question: is this repo only for command real go2 or also for the simulated one in gazebo?

**Comment by AsdoubleU** [comment]
Just for the Robot in the Gazebo Simulation. Not for the real robot.

**Comment by AsdoubleU** [comment]
My repo is for the only for the robot in the Gazebo simulation, and the link what I introduced you is only for the real go2 robot. You should read unitree website. Have a nice day~!

**Comment by Aleksej9521** [comment]
Okay, thanks :) I was trying to run  the go2_torque but I have this kind of error 
![Schermata da 2024-04-28 11-24-04](https://github.com/AsdoubleU/Go2_Gazebo_Environment/assets/73386894/f1f58276-4234-4339-872a-dfc4b2a5adfb). Have you ever encountered this problem?

**Comment by AsdoubleU** [comment]
In my opinion, those errors are kind of matrix errors. You should check the matrix multiplications or the version of Eigen.

**Comment by Aleksej9521** [comment]
Okay. What version of Eigen did you use? I downloaded the 3.3.1. I found out that the problem its related to "SetRobotParameter" function in the pinocchio _interface.cpp, specifically in the calculation of ddq (as in the picture I attach). 
![immagine](https://github.com/AsdoubleU/Go2_Gazebo_Environment/assets/73386894/e3587dab-828a-4baa-a33e-b321d1e8a462). I tried with changing the version of Eigen, but I still have the same error.
